template/runtime/src/lib.rs

```diff
@@ -312,11 +315,36 @@ impl<F: FindAuthor<u32>> FindAuthor<H160> for FindAuthorTruncated<F> {
 	}
 }
 
-const WEIGHT_PER_GAS: u64 = 20_000;
+/// `WeightPerGas` is an approximate ratio of the amount of Weight per Gas.
+/// u64 works for approximations because Weight is a very small unit compared to gas.
+///
+/// `GAS_PER_MILLIS * WEIGHT_MILLIS_PER_BLOCK * TXN_RATIO ~= BLOCK_GAS_LIMIT`
+/// `WEIGHT_PER_GAS = WEIGHT_REF_TIME_PER_MILLIS / GAS_PER_MILLIS
+///                 = WEIGHT_REF_TIME_PER_MILLIS / (BLOCK_GAS_LIMIT / TXN_RATIO / WEIGHT_MILLIS_PER_BLOCK)
+///                 = TXN_RATIO * (WEIGHT_REF_TIME_PER_MILLIS * WEIGHT_MILLIS_PER_BLOCK) / BLOCK_GAS_LIMIT`
+///
+/// For example, given the 2000ms Weight, from which 75% only are used for transactions,
+/// the total EVM execution gas limit is `GAS_PER_MILLIS * 2000 * 75% = BLOCK_GAS_LIMIT`.
+pub fn weight_per_gas(
+	block_gas_limit: u64,
+	txn_ratio: Perbill,
+	weight_millis_per_block: u64,
+) -> u64 {
+	let weight_per_block = WEIGHT_REF_TIME_PER_MILLIS.saturating_mul(weight_millis_per_block);
+	let weight_per_gas = (txn_ratio * weight_per_block).saturating_div(block_gas_limit);
+	assert!(
+		weight_per_gas >= 1,
+		"WeightPerGas must greater than or equal with 1"
+	);
+	weight_per_gas
+}
+
+const BLOCK_GAS_LIMIT: u64 = 75_000_000;
```

Comment:
How did you get this `BLOCK_GAS_LIMIT`? There seems to be a dead loop in your solution. When you calculate the `BLOCK_GAS_LIMIT`,  you also need `WeightPerGas`.

Reply:
if not that, how did you get the WEIGHT_PER_GAS?

Reply:
I remembered that the const value was estimated by the benchmark. 
https://github.com/paritytech/frontier/blob/master/frame/evm/src/benchmarking.rs

Reply:
I'm working on 994, an issue that has some relevance to the current pr.

> previously we get block gas limit from weight per gas.

`block_gas_limit` was previously calculated from `block_max_weight` and `weight_per_gas`.  `block_max_weight` is known in runtime and `weight_per_gas` can be obtained from pallet_evm's benchmark.

block_gas_limit = block_max_weight / weight_per_gas.

> one can get a much more accurate estimate of the overall block gas limit through benchmarking (because it's aggregated)

Can you go a bit deeper into your idea? @sorpaas 

Benchmark the maximum value of transaction gas that a block of fixed weight size can hold? That doesn't seem to work, the Ethereum::transact() call's weight calculation requires `weight_per_gas`, and the `weigth_per_gas` calculation requires `block_gas_limit`, which is like a loop.